pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -213,20 +248,157 @@ func (c *MCDefaultRouteController) processNextWorkItem() bool {
+//     configuration and route on the host, then add all existing ClusterInfoImports as WireGuard peers.
+//  2. If the current Node is not Multi-cluster Gateway Node, controller will try to clean up WireGuard.
+//
+// Note: MCDefaultRouteController runs only one worker to process Gateway and ClusterInfoImport. So we do not need
+// any synchronization mechanism.
+func (c *MCDefaultRouteController) reconcileWireGuard() error {
+	gateway, err := c.getActiveGateway()
+	if err != nil {
+		return err
+	}
+
+	amIGateway := gateway != nil && gateway.Name == c.nodeConfig.Name
+	if !amIGateway {
+		if c.wireGuardInitialized {
+			if err := c.cleanUpWireGuard(); err != nil {
+				return err
+			}
+			c.wireGuardInitialized = false
+		}
+		return nil
+	}
+	if !c.wireGuardInitialized {
+		if err := c.initializeWireGuard(); err != nil {
+			return err
+		}
+		c.wireGuardInitialized = true
+	}
+	ciImports, _ := c.ciImportLister.List(labels.Everything())
+	for _, ciImport := range ciImports {
+		if err := c.updateSingleWireGuardPeer(ciImport); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// updateSingleWireGuardPeer tries to update a WireGuard peer with ClusterInfoImport. If updating successfully,
+// it will also create host route the peer.
+func (c *MCDefaultRouteController) updateSingleWireGuardPeer(ciImport *mcv1alpha1.ClusterInfoImport) error {
+	klog.InfoS("Updating WireGuard peer with ClusterInfoImport", "ClusterInfoImport", ciImport)
+	if ciImport.Spec.WireGuard == nil || ciImport.Spec.WireGuard.PublicKey == "" {
+		klog.InfoS("ClusterInfoImport's WireGuard field has not been initialized, skip it", "ClusterInfoImport", ciImport)
+		return nil
+	}
+	serviceIP, serviceCIDR, err := net.ParseCIDR(ciImport.Spec.ServiceCIDR)
+	if err != nil {
+		return err
+	}
+	remoteWireGuardIP := serviceIP.Mask(serviceCIDR.Mask)
+	_, remoteWireGuardNet, err := net.ParseCIDR(remoteWireGuardIP.String() + "/32")
+	if err != nil {
+		return err
+	}
+
+	gatewayIP := net.ParseIP(ciImport.Spec.GatewayInfos[0].GatewayIP)
+	allowedIPs := []*net.IPNet{remoteWireGuardNet}
+	if err := c.wireGuardClient.UpdatePeer(ciImport.Name, ciImport.Spec.WireGuard.PublicKey, gatewayIP, allowedIPs); err != nil {
+		klog.ErrorS(err, "Failed to update WireGuard peer", "ClusterInfoImport", ciImport)
+		return err
+	}
+	klog.InfoS("Updated WireGuard peer with ClusterInfoImport", "ClusterInfoImport", ciImport.Name)
+	if err := c.addWireGuardRoute(remoteWireGuardIP.String()); err != nil {
+		klog.ErrorS(err, "Failed to add WireGuard route on the host", "remoteGateway", ciImport.Name)
+		return err
+	}
+	return nil
+}
+
+// initializeWireGuard initializes the WireGuard interface and client.
+// It will also update Gateway's WireGuard field.
+func (c *MCDefaultRouteController) initializeWireGuard() error {
+	c.wireGuardConfig.MTU = c.nodeConfig.NodeTransportInterfaceMTU - c.networkConfig.MTUDeduction - config.WireGuardOverhead
```

Comment:
I think you can initialize this MTU in `NewMCDefaultRouteController` or in the cmd/antrea-agent/options.go?

Reply:
Moved to `NewMCDefaultRouteController`.